### ISSUE TITLE
NPM Install <package> overrides/deletes Git repositories listed in package.json

### ISSUE THREAD

**Issue by carboneater** [npm/npm#19945]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ *] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
We use NPM as a package manager using both the npm registry and our private Git server.

When installing a package with `npm install <package>` instruction, NPM uninstalls all the git packages listed in _package.json_ prior to installing the named package.
Running `npm install` afterwards with no arguments reinstalls all the packages listed in package.json.

Here's the result of my last run:
```powershell
PS > npm install --save moment-timezone
+ moment-timezone@0.5.14
added 1 package and removed 125 packages in 32.244s
PS > npm install --save-dev @types/moment-timezone
+ @types/moment-timezone@0.5.4
added 1 package in 1.318s
PS > npm install
added 125 packages in 29.751s
```

#### How can the CLI team reproduce the problem?
1. Create a new project
2. Insert a package.json containing private git repositories (git+https protocol)
3. Run `npm install`
Install works fine.
4. run `npm install <new_package>`
Npm uninstalls all git-provided packages.

Npm does not print any _npm-debug.log_.

### supporting information:

 - `npm -v` prints: *5.6.0*
 - `node -v` prints: _v8.9.4_
 - `npm config get registry` prints: *https://registry.npmjs.org/*
 - Windows, OS X/macOS, or Linux?: Windows over PowerShell and MinGW
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by G-Rath** [comment]
This is a duplicate of #19394, and has supposedly been fixed in npm@5.7

**Comment by carboneater** [comment]
Indeed, updating to npm 5.7.1 seems to fix the issue.

Thanks for the heads up!
